pkg/autoscaler/autoscaler.go

```diff
@@ -877,4 +895,130 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
+			framework.WaitForMachineSet(ctx, client, transientMachineSets[0].GetName())
+			framework.WaitForMachineSet(ctx, client, transientMachineSets[1].GetName())
+
+			maxMachineSetReplicas := int32(2)
+			for _, machineSet := range transientMachineSets {
+				By(fmt.Sprintf("Creating a MachineAutoscaler backed by MachineSet %s - min: 1, max: %d",
+					machineSet.GetName(), maxMachineSetReplicas))
+				asr := machineAutoscalerResource(machineSet, 1, maxMachineSetReplicas)
+				Expect(client.Create(ctx, asr)).Should(Succeed(), "Failed to create MachineAutoscaler with min 1/max %d replicas", maxMachineSetReplicas)
+				cleanupObjects[asr.GetName()] = asr
+			}
+
+			By("Create cluster-autoscaler-priority-expander")
+			priorities := "10: \n  - .*" + transientMachineSets[0].GetName() + ".*\n20:\n  - .*" + transientMachineSets[1].GetName() + ".*"
+			cm := configMapResource(priorities)
+			Expect(client.Create(ctx, cm)).Should(Succeed(), "Failed to create ConfigMap with priorities %s", priorities)
+			cleanupObjects[cm.GetName()] = cm
+
+			jobReplicas := int32(4)
+			uniqueJobName := fmt.Sprintf("%s-priority-expander", workloadJobName)
+			By(fmt.Sprintf("Creating scale-out workload %s: jobs: %v, memory: %s",
+				uniqueJobName, jobReplicas, workloadMemRequest.String()))
+			workload := framework.NewWorkLoad(jobReplicas, workloadMemRequest, uniqueJobName, autoscalingTestLabel, "", corev1.NodeSelectorRequirement{
+				Key:      targetedNodeLabel,
+				Operator: corev1.NodeSelectorOpExists,
+			})
+			cleanupObjects[workload.GetName()] = workload
+			Expect(client.Create(ctx, workload)).Should(Succeed(), "Failed to create scale-out workload %s", uniqueJobName)
+
+			By("Check machinesets scale up to 2 machines")
+			expectedReplicas := int32(2)
+			Eventually(func() bool {
+				ms, err := framework.GetMachineSet(ctx, client, transientMachineSets[0].GetName())
+				Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSets %s", transientMachineSets[0].GetName())
+				ms1, err := framework.GetMachineSet(ctx, client, transientMachineSets[1].GetName())
+				Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSets %s", transientMachineSets[1].GetName())
+
+				By(fmt.Sprintf("Waiting for machineSets replicas to scale out. Current replicas for machineset %s are %v, expected %v, for machineset %s are %v, expected %v .",
+					transientMachineSets[0].GetName(), *ms.Spec.Replicas, expectedReplicas, transientMachineSets[1].GetName(), *ms1.Spec.Replicas, expectedReplicas))
+
+				return *ms.Spec.Replicas == expectedReplicas && *ms1.Spec.Replicas == expectedReplicas
+			}, framework.WaitMedium, pollingInterval).Should(BeTrue(), "MachineSet %s failed to scale out to %d replicas or MachineSet %s failed to scale out to %d replicas", transientMachineSets[0].GetName(), expectedReplicas, transientMachineSets[1].GetName(), expectedReplicas)
+
+			By("Check autoscaler scales up based on Priority")
+			Eventually(func() bool {
```

Comment:
Does this actually need to be wrapped in an eventually? Given we have already scaled up and checked the expected replicas at this point?

I think we could pull this out and then use a gomega matcher to compare the timestamps